### PR TITLE
Add general project direction and user stories for DG

### LINKED ISSUE #77: Add general project direction and user stories for DG
Add target user profile, value proposition, and user stories

### PR DESCRIPTION
**Pull request by joeltio**
Closes #77

The target user profile is taken from the main Google document and adapted to fit to the developer guide format. As you can see, nothing much is changed, just providing some context to the given user profile.

User stories are copied from the Consolidated User Stories sheet with some slight modification to the wording for the "So that I can..." column.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -262,27 +262,50 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `* *`    | sales rep        | add tags to clients                                                         | categorize them                                                   |
+| `* *`    | sales rep        | keep note of my client's email addresses                                    | potentially send promotions or survey forms                       |
+| `* *`    | sales rep        | export a list of contact emails                                             | add them to a mailing list                                        |
+| `* *`    | sales rep        | add a tag to multiple clients                                               | tag the clients more easily                                       |
+| `*`      | experienced user | use keyboard shortcuts                                                      | navigate the app faster                                           |
+| `*`      | sales rep        | contact my client quickly from the app                                      | avoid typing numbers repeatedly on my phone                       |
```
Comment (KennethTeo2002) [inline docs/DeveloperGuide.md:308]: Shouldn't this be desktop app? Since it is stated In the target user profile.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -262,27 +262,50 @@ _{Explain here how the data archiving feature will be implemented}_
...
-* has a need to manage a significant number of contacts
+A sales and customer relations representative working in the F&B industry. In
+particular, this representative works with B2B sales.
+
+* has a need to manage a significant number of business contacts
 * prefer desktop apps over other types
 * can type fast
 * prefers typing to mouse interactions
 * is reasonably comfortable using CLI apps
+* has a need to transfer large number of contacts to new workers
```
Comment (KennethTeo2002) [inline docs/DeveloperGuide.md:273]: Not sure if this should be part of user profile.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -262,27 +262,50 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `* *`    | sales rep        | add tags to clients                                                         | categorize them                                                   |
+| `* *`    | sales rep        | keep note of my client's email addresses                                    | potentially send promotions or survey forms                       |
+| `* *`    | sales rep        | export a list of contact emails                                             | add them to a mailing list                                        |
+| `* *`    | sales rep        | add a tag to multiple clients                                               | tag the clients more easily                                       |
+| `*`      | experienced user | use keyboard shortcuts                                                      | navigate the app faster                                           |
+| `*`      | sales rep        | contact my client quickly from the app                                      | avoid typing numbers repeatedly on my phone                       |
 
 *{More to be added}*
```
Comment (KennethTeo2002) [inline docs/DeveloperGuide.md:310]: Should we remove this placeholder?

**Review by KennethTeo2002 — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Nicholas-Cheng-De-Fei) [reply docs/DeveloperGuide.md:310]: I think it is alright to leave it since Sheen Kerr will be updating this for his part since he is doing the bring over of the user stories?

**Review by Nicholas-Cheng-De-Fei — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (joeltio) [reply docs/DeveloperGuide.md:308]: It is a desktop app. The implementation of this could be integration with WhatsApp Web/Telegram Web or some sort of VoIP.

Of course, we won't be doing it in this project. I can remove it if we decide that we want the user stories to represent things that we are more likely to actually do in the project.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (joeltio) [reply docs/DeveloperGuide.md:273]: I added this incorrectly. I was intending to add this to justify for the user story below about "export a list of contact emails".

Will remove.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -262,27 +262,50 @@ _{Explain here how the data archiving feature will be implemented}_
...
+* has a need to manage a significant number of business contacts
 * prefer desktop apps over other types
 * can type fast
 * prefers typing to mouse interactions
 * is reasonably comfortable using CLI apps
+* has a need to transfer large number of contacts to new workers
 
-**Value proposition**: manage contacts faster than a typical mouse/GUI driven app
+**Value proposition**: manage and organise large number of contacts faster than a
+typical mouse/GUI driven app
```
[the remark this answers is not in the document]
Comment (joeltio) [reply docs/DeveloperGuide.md:276]: I can rephrase it to add in the word efficiency though I think it is shown through speed ("faster") and well..I wouldn't say a CLI-based application is easy-to-use...

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (joeltio) [reply docs/DeveloperGuide.md:310]: Hmm I left it as I was thinking if we are going to have more user stories. But I think we will be finalising probably by this sprint. Will remove.

**Review by Nicholas-Cheng-De-Fei — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -266,28 +266,51 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `* *`    | user             | search through my contacts                                                  | find a specific person                                                 |
+| `* *`    | new user         | see usage instructions                                                      | know how to use the app                                                |
+| `* *`    | user             | edit contact                                                                | update contact with new information                                    |
+| `* *`    | user             | sort contact by name                                                        | see whose contact I have saved                                         |
+| `* *`    | user             | archive contact                                                             | hide less frequently used contacts without deleting them               |
+| `* *`    | user             | be alerted when a contact already exist                                     | avoid accidentally creating a duplicate                                |
```
Comment (KennethTeo2002) [inline docs/DeveloperGuide.md:298]: I dont think this is a valid use case? The user is not trying to achieve any action.

**Review by KennethTeo2002 — CHANGES_REQUESTED** [review]

**Review by KennethTeo2002 — APPROVED** [review]
LGTM
